src/autocomplete.js

```diff
@@ -30,16 +42,34 @@ var Autocomplete = function() {
 };
 
 (function() {
+    this.$onPopupChange = function(hide) {
+        if (hide) {
+            if (this.inlineEnabled) {
+                this.inlineRenderer.show(this.editor, null, prefix);    
+            }
+            this.hideDocTooltip();
+        } else {
+            this.tooltipTimer.call(null, null);
+            if (this.inlineEnabled) {
+                var completion = hide ? null : this.popup.getData(this.popup.getRow());
+                var prefix = util.getCompletionPrefix(this.editor);
+                this.inlineRenderer.show(this.editor, completion, prefix);
+                this.$updatePopupPosition();
+            }
+        }
+    };
 
     this.$init = function() {
         this.popup = new AcePopup(document.body || document.documentElement);
+        this.inlineRenderer = new AceInline();
```

Comment:
Do we always need this or only when inlineEnabled is true?

Reply:
It is only needed when inlineEnabled is true, I'll make sure it's not created needlessly.